packages/block-editor/src/store/selectors.js

```diff
@@ -1435,6 +1435,10 @@ export function getTemplateLock( state, rootClientId ) {
 		return state.settings.templateLock;
 	}
 
+	if ( getBlockAttributes( state, rootClientId ).lock?.content === true ) {
+		return 'all';
+	}
+
```

Comment:
Looking at this change, I wonder if it makes sense for content locking to be a part of `templateLock`, instead of block locking.

Reply:
Good question 👍 Yes, I guess we should consider making it a templateLock = content. It seems to make sense.

Reply:
I've been educating myself about the different locking mechanism we have. I agree that using `templateLock` makes more sense than `lock` because `templateLock` aims to control the inner blocks behavior while `lock` controls the block itself.

I see some challenges:

- Naming. We should follow the existing naming schema and indicating _what_ it locks, not what it _allows_ to edit. `noContent`  _could_ work.
- Semantics. Introducing a new state to `templateLock` that is not absorved by the existing `all` is going to create disonance for users of the API. Example:
	- inserts => locks insertion/removal
	- all => "inserts" plus locks movement
	- noContent => "all" plus a bunch of other things (prevent selection for blocks without content, hide them listview, absorb the block inspector of the inner blocks, etc)
- Nested templateLock. The existing locking mechanism fallbacks through a hierarchy: PostType's templateLock < an InnerBlocks's templateLock < another InnerBlocks's templateLock < a singe Block's lock. The latest has more priority. Does the new `noContent` state also flows through the hierarchy? If so, what's the expected behavior here:

```html
<group templateLock=noContent>
  <group templateLock=all />
</group>
```

- Following the reasoning above, there's an existing  `templateLock=false` state. If a block has this state, it "unlocks" anything defined up in the hierarchy. Some blocks use it: social links, navigation, and column. If we respect the cascading behaviour and `false` can unlock anything, the new state won't be effective for those blocks.
- `templateLock` and `lock` and interrelated. We should consider introducing the corresponding states for `lock` if we introduce one for `templateLock`.

Chewing about it, this is the next steps I see:

- Update this PR to use a new state in templateLock instead of lock. Despite the challenges it makes more sense.
- Ignore the value of templateLock for nested containers in this PR. For controlling scope, it should be addressed in a follow-up.
- We may need to consider exposing `templateLock` as an attribute of colum, navigation, and social links should the `false` state needs to be overriden by a pattern. To be tested and investigated when addressing nested content.
 - I'm unsure about introducing a new state for lock. It makes sense conceptually, though I can't see now any practical uses for it. Punt it for the future if need be.

Thoughts?

Reply:
Pushed changes to substitute `lock.content` by `templateLock: "noContent"`. It should work the same as before.